src/Nethermind/Nethermind.Flashbots/Handlers/ValidateBuilderSubmissionHandler.cs

```diff
@@ -0,0 +1,432 @@
+// SPDX-FileCopyrightText: 2024 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Nethermind.Blockchain;
+using Nethermind.Blockchain.BeaconBlockRoot;
+using Nethermind.Blockchain.Blocks;
+using Nethermind.Blockchain.Receipts;
+using Nethermind.Flashbots.Data;
+using Nethermind.Consensus;
+using Nethermind.Consensus.Processing;
+using Nethermind.Consensus.Validators;
+using Nethermind.Consensus.Withdrawals;
+using Nethermind.Core;
+using Nethermind.Core.Specs;
+using Nethermind.Crypto;
+using Nethermind.Evm;
+using Nethermind.Evm.Tracing;
+using Nethermind.Evm.TransactionProcessing;
+using Nethermind.Int256;
+using Nethermind.JsonRpc;
+using Nethermind.Logging;
+using Nethermind.Merge.Plugin.Data;
+using Nethermind.State;
+using Nethermind.Core.Crypto;
+
+namespace Nethermind.Flashbots.Handlers;
+
+public class ValidateSubmissionHandler
+{
+    private ProcessingOptions ValidateSubmissionProcessingOptions = ProcessingOptions.ReadOnlyChain
+         | ProcessingOptions.IgnoreParentNotOnMainChain
+         | ProcessingOptions.ForceProcessing
+         | ProcessingOptions.StoreReceipts;
+    private readonly ReadOnlyTxProcessingEnv _txProcessingEnv;
+
+    private readonly ReadOnlyTxProcessingEnvFactory _readOnlyTxProcessingEnvFactory;
+
+    private readonly IBlockTree _blockTree;
+    private readonly IHeaderValidator _headerValidator;
+    private readonly IBlockValidator _blockValidator;
+    private readonly ILogger _logger;
+
+    private readonly IFlashbotsConfig _flashbotsConfig;
+
+    private readonly IReceiptStorage _receiptStorage = new InMemoryReceiptStorage();
```

Comment:
Do we need receipt storage at all? Maybe `NullReceiptStorage`?

Reply:
No we don't need it anymore, I have removed it as well